lib/Runtime/PlatformAgnostic/Platform/Linux/DateTime.cpp

```diff
@@ -13,155 +13,219 @@ namespace PlatformAgnostic
+
+        // mktime etc. broken-out time accepts 1900 as a start year while epoch is 1970
+        // minus the previously added calendar day (see YMD_TO_TM)
+        if (leap_added)
+        {
+            AssertMsg(ymd->mday >= 0, "Day of month can't be a negative number");
+            if (ymd->mday == 0)
+            {
+                ymd->mday = 29;
+                ymd->mon = 1;
+            }
+            else
+            {
+                ymd->mday--;
+            }
         }
     }
 
-    // DateTimeHelper ******
-    static inline void TM_TO_SYSTIME(struct tm *time, SYSTEMTIME *sysTime)
+    static void CopyTimeZoneName(WCHAR *wstr, size_t *length, const char *tm_zone)
     {
-        sysTime->wYear = time->tm_year + 1900;
-        sysTime->wMonth = time->tm_mon + 1;
-        sysTime->wDayOfWeek = time->tm_wday;
-        sysTime->wDay = time->tm_mday;
-        sysTime->wHour = time->tm_hour;
-        sysTime->wMinute = time->tm_min;
-
-        // C99 tm_sec value is between 0 and 60. (1 leap second.)
-        // Discard leap second.
-        // In any case, this is a representation of a wallclock time.
-        // It can jump forwards and backwards.
-
-        sysTime->wSecond = time->tm_sec % 60;
+        *length = strlen(tm_zone);
+
+#if defined(WCHAR_IS_CHAR16_T)
+        for(int i = 0; i < *length; i++) {
+            wstr[i] = (char16_t)tm_zone[i];
+        }
+#elif defined(WCHAR_IS_WCHAR_T)
+        mbstowcs( (wchar_t*) wstr, tm_zone, __CC_PA_TIMEZONE_ABVR_NAME_LENGTH);
+#else
+        #error "WCHAR should be either wchar_t or char16_t"
+#endif
+
+        wstr[*length] = (WCHAR)0;
     }
 
-    #define MIN_ZERO(value) value < 0 ? 0 : value
+    const WCHAR *Utility::GetStandardName(size_t *nameLength, const DateTime::YMD *ymd) {
+        AssertMsg(ymd != NULL, "xplat needs DateTime::YMD is defined for this call");
+        struct tm time_tm;
+        bool leap_added;
+        YMD_TO_TM(ymd, &time_tm, &leap_added);
+        mktime(&time_tm); // get zone name for the given date
+        CopyTimeZoneName(data.standardName, &data.standardNameLength, time_tm.tm_zone);
+        *nameLength = data.standardNameLength;
+        return data.standardName;
+    }
 
-    static inline void SYSTIME_TO_TM(SYSTEMTIME *sysTime, struct tm *time)
-    {
-        time->tm_year = MIN_ZERO(sysTime->wYear - 1900);
-        time->tm_mon = MIN_ZERO(sysTime->wMonth - 1);
-        time->tm_wday = sysTime->wDayOfWeek;
-        time->tm_mday = sysTime->wDay;
-        time->tm_hour = sysTime->wHour;
-        time->tm_min = sysTime->wMinute;
-        time->tm_sec = sysTime->wSecond;
+    const WCHAR *Utility::GetDaylightName(size_t *nameLength, const DateTime::YMD *ymd) {
+        // xplat only gets the actual zone name for the given date
+        return GetStandardName(nameLength, ymd);
     }
 
-    static void SysLocalToUtc(SYSTEMTIME *local, SYSTEMTIME *utc)
+    static void SysLocalToUtc(YMD *local, YMD *utc)
```

Comment:
Sys over here in SysLocalToUtc actually referes to SYSTEMTIME. Since you are now passing in YMD, you should change this to YmdLocalToUtc